src/controllers/commands/CommandController.cpp

```diff
@@ -1355,6 +1355,126 @@ void CommandController::initialize(Settings &, Paths &paths)
             (void)words;  // unused
             return deleteMessages(channel, QString());
         });
+
+    this->registerCommand("/mod", [](const QStringList &words, auto channel) {
+        if (words.size() < 2)
+        {
+            channel->addMessage(makeSystemMessage(
+                "Usage: \"/mod <username>\" - Grant moderator status to a "
+                "user. Use \"/mods\" to list the moderators of this channel."));
+            return "";
+        }
+
+        auto currentUser = getApp()->accounts->twitch.getCurrent();
+        if (currentUser->isAnon())
+        {
+            channel->addMessage(
+                makeSystemMessage("You must be logged in to mod someone!"));
+            return "";
+        }
+
+        auto *twitchChannel = dynamic_cast<TwitchChannel *>(channel.get());
+        if (twitchChannel == nullptr)
+        {
+            channel->addMessage(makeSystemMessage(
+                "The /mod command only works in Twitch channels"));
+            return "";
+        }
+
+        if (twitchChannel->roomId() != currentUser->getUserId())
+        {
+            channel->addMessage(makeSystemMessage(
+                "You may only mod people in your own channel."));
+            return "";
+        }
+
+        auto target = words.at(1);
+        stripChannelName(target);
+
+        getHelix()->getUserByName(
+            target,
+            [currentUser, channel](const HelixUser &targetUser) {
+                getHelix()->addChannelModerator(
+                    currentUser->getUserId(), targetUser.id,
+                    [channel, targetUser] {
+                        channel->addMessage(makeSystemMessage(
+                            QString("You have added %1 as a moderator of this "
+                                    "channel.")
+                                .arg(targetUser.displayName)));
+                    },
+                    [channel, targetUser](auto error, auto message) {
+                        QString errorMessage =
+                            QString("Failed to add channel moderator - ");
+
+                        switch (error)
+                        {
+                            case HelixAddChannelModeratorError::
```

Comment:
You can use `using Error = HelixAddChannelModeratorError;` in this lambda' function body to make the error-switch a bit more readable

Reply:
fixed

Reply:
not fixed :fearful:

Reply:
oh my bad, wrong file lol, now its fixed